modules/programs/alot-accounts.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+  options.alot = {
+    sendMailCommand = mkOption {
+      type = types.nullOr types.str;
+      description = ''
+        Command to send a mail. If msmtp is enabled for the account,
+        then this is set to
+        <command>msmtpq --account=&lt;name&gt; -t</command>.
+      '';
+    };
+  };
+
+  config = mkIf config.notmuch.enable {
+    alot.sendMailCommand = mkOptionDefault (
+      if config.msmtp.enable
+      then "msmtpq --account=${config.name} -t"
```

Comment:
while using my astroid and alot modules, I realized it makes more sense to `msmtpq --read-envelope-from --read-recipients`. With multiple accounts, both alot and astroid will let you choose interactively the sender address. When using --read-envelope-from, msmtp will choose the appropriate account whereas msmtpq --account forces always the same account.